website/docs/r/tenant_policies_dhcp_relay_policy.html.markdown

```diff
@@ -0,0 +1,58 @@
+---
+layout: "mso"
+page_title: "MSO: mso_tenant_policies_dhcp_relay_policy"
+sidebar_current: "docs-mso-resource-tenant_policies_dhcp_relay_policy"
+description: |-
+  Manages DHCP Relay Policies on Cisco Nexus Dashboard Orchestrator (NDO)
+---
+
+# mso_tenant_policies_dhcp_relay_policy #
+
+Manages DHCP Relay Policies on Cisco Nexus Dashboard Orchestrator (NDO). This resource is supported in NDO v4.4(1) or higher.
+
+## GUI Information ##
+
+* `Location` - Manage -> Tenant Template -> Tenant Policies -> DHCP Relay Policy
+
+## Example Usage ##
+
+```hcl
+resource "mso_tenant_policies_dhcp_relay_policy" "dhcp_relay_policy" {
+  name        = "dhcp_relay_policy"
+  template_id = mso_template.tenant_policy_template.id
+  description = "example_dhcp_relay_policy"
+  providers {
+    dhcp_server_address  = "1.1.1.1"
+    application_epg_uuid = mso_schema_template_anp_epg.anp_epg.uuid
+  }
+  providers {
+    dhcp_server_address        = "2.2.2.2"
+    external_epg_uuid          = mso_schema_template_external_epg.ext_epg.uuid
+    dhcp_server_vrf_preference = true
+  }
+}
+```
+
+## Argument Reference ##
+
+* `template_id` - (Required) The unique ID of the template.
+* `name` - (Required) The name of the DHCP relay policy.
+* `description` - (Optional) The description of the DHCP relay policy.
+* `providers` - (Required) A list of providers for the DHCP relay policy.
```

Comment:
I am afraid that this would be confused with https://developer.hashicorp.com/terraform/language/meta-arguments/resource-provider and maybe even conflict in the future. I think we might want to change it to:
 ```suggestion
* `relay_providers` - (Required) A list of providers for the DHCP relay policy.
```
or
```suggestion
* `dhcp_relay_providers` - (Required) A list of providers for the DHCP relay policy.
```

Reply:
I agree with @lhercot, from terraform ACI provider I also noticed that provider is a reserved schema attribute. I do not think we would have an issue because I have not seen any plural forms in HCL, but do agree this might be conflicting in the future. I would say `dhcp_relay_providers` in this case to make it as clear as possible.

Reply:
Done!